crates/task-impls/src/helpers.rs

```diff
@@ -42,15 +42,38 @@ pub async fn broadcast_event<E: Clone + std::fmt::Debug>(event: E, sender: &Send
 ///
 /// # Panics
 /// Panics if the VID calculation fails, this should not happen.
+#[allow(clippy::panic)]
 pub async fn calculate_vid_disperse<TYPES: NodeType>(
     txns: Vec<u8>,
     membership: Arc<TYPES::Membership>,
     view: TYPES::Time,
 ) -> VidDisperse<TYPES> {
     let num_nodes = membership.total_nodes();
     let vid_disperse = spawn_blocking(move || {
-        #[allow(clippy::panic)]
-        vid_scheme(num_nodes).disperse(&txns).unwrap_or_else(|err|panic!("VID disperse failure:\n\t(num_storage nodes,payload_byte_len)=({num_nodes},{})\n\terror: : {err}", txns.len()))
+        vid_scheme(num_nodes).disperse(&txns).unwrap_or_else(|err| panic!("VID disperse failure:(num_storage nodes,payload_byte_len)=({num_nodes},{}) error: {err}", txns.len()))
+    })
+    .await;
+    #[cfg(async_executor_impl = "tokio")]
```

Comment:
I have questions. I don't think there's any other point in the HotShot repo where we hardcode only tokio. Maybe it would be worth unravelling this spawn_blocking just to avoid that.

Reply:
https://github.com/EspressoSystems/HotShot/pull/2956/commits/5408f863bc99171c7e299ac0460875145d98252d